deepchecks/vision/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains the domain classifier drift check."""
+from collections import OrderedDict
+from typing import Any, List
+
+from deepchecks.core import CheckResult, DatasetKind
+from deepchecks.vision import Context, TrainTestCheck
+from deepchecks.core.check_utils.whole_dataset_drift_utils import run_whole_dataset_drift
+import pandas as pd
+
+__all__ = ['WholeDatasetDrift']
+
+DEFAULT_IMAGE_PROPERTIES = ['aspect_ratio',
+                            'area',
+                            'brightness',
+                            'normalized_red_mean',
+                            'normalized_green_mean',
+                            'normalized_blue_mean']
+
+
+class WholeDatasetDrift(TrainTestCheck):
+    """
+    Calculate drift between the entire train and test datasets using a model trained to distinguish between them.
+
+    Check fits a new model to distinguish between train and test datasets, called a Domain Classifier.
```

Comment:
Worth adding a few words of "why?" - why is this check useful for me and when should I run it,

Reply:
I love it when you comment on basically your own text :)
But seriously - this is a lot of text anyway, and I don't see us "justify" when to use a check in other places, so not sure what you mean here

Reply:
You may say I'm commenting on stuff you didn't catch when reviewing my text 😉 
Anyway, yeah it suddenly bothered me, but if we're not doing it in other checks then no sense starting here.